File: src/main/java/com/dsa/problems/scaler/bitwise_Operators/addition.java
package com.dsa.problems.scaler.bitwise_Operators;

public class addition {
  /**
   * Implement a function to add two numbers using bitwise operators.
   */
  public static int add(int A, int B) {
    while(B != 0) {
      int carry = A & B;
      A = A ^ B;
      B = carry << 1;
    }
    return A;
  }

  public static void main(String[] args) {
    System.out.println(add(1, 2));
    System.out.println(add(10, 20));
  }
}

File: problems/scaler/Bitwise_operators/addition.js
/**
 * Implement Addition operation using bitwise operators
 */
function soln(A, B) {
  while (B !== 0) {
    // Calculate carry
    let carry = A & B;

    // Calculate sum without carry
    A = A ^ B;

    // Shift carry to left by 1 position
    B = carry << 1;
  }

  return A;
}

console.log(soln(1, 2)); // 3
console.log(soln(10, 15)); // 25
console.log(soln(23, 45)); // 68
